### ISSUE TITLE
The bandwidth consumption is large

### ISSUE THREAD

**Issue by jiaokele** [paulov-t/SIT.Core#92]
![image](https://github.com/paulov-t/SIT.Core/assets/32627730/3db9c5c2-4274-418b-887d-86454c3ed551)

Four to six people are playing at the same time. The traffic consumption is large, and the delay is also large.
Duplicate data may exist.
Or can you reduce the type of data you transfer?
Or can the data be compressed?
Or just the critical data.

Here are some of the things we found on the tour.
![image](https://github.com/paulov-t/SIT.Core/assets/32627730/5845840a-9497-409a-9f35-d5b8ed427ed1)

**Comment by SlejmUr** [comment]
> Or can you reduce the type of data you transfer?

No, maybe more compressed or smth but , as I am aware the data is send by BSG/AKI's solution of ZLIB so it sending compressed packets.
Because of the mod is still early ye, the bandwith, and all the things will improve in the future if the basics are done.

> Or can the data be compressed?

Check back to first answer, data is send with compression.

> Or just the critical data.

Because all the packets communicating with TCP you cannot send "Unreliable" packets as in UDP.
Paulov tested udp a way back and it wasnt the best solution (as he says)

**Comment by w123456w30w** [comment]
> > Or can you reduce the type of data you transfer?
> 
> No, maybe more compressed or smth but , as I am aware the data is send by BSG/AKI's solution of ZLIB so it sending compressed packets. Because of the mod is still early ye, the bandwith, and all the things will improve in the future if the basics are done.
> 
> > Or can the data be compressed?
> 
> Check back to first answer, data is send with compression.
> 
> > Or just the critical data.
> 
> Because all the packets communicating with TCP you cannot send "Unreliable" packets as in UDP. Paulov tested udp a way back and it wasnt the best solution (as he says)



> > Or can you reduce the type of data you transfer?
> 
> No, maybe more compressed or smth but , as I am aware the data is send by BSG/AKI's solution of ZLIB so it sending compressed packets. Because of the mod is still early ye, the bandwith, and all the things will improve in the future if the basics are done.
> 
> > Or can the data be compressed?
> 
> Check back to first answer, data is send with compression.
> 
> > Or just the critical data.
> 
> Because all the packets communicating with TCP you cannot send "Unreliable" packets as in UDP. Paulov tested udp a way back and it wasnt the best solution (as he says)

Hey, I have a question. If the server is acting as a relay between the host and the clients, in the case of two players, is the connection behavior "Host <-> Server <-> Client?" right? 
What about multiple clients? the server will cache the same information to broadcast to other clients?

**Comment by SlejmUr** [comment]
C1=>S=>H,C2; 
C2=>S=>H,C1; 
H=>S=>C1,C2

Yes the server is always work as a relay between players

**Comment by SlejmUr** [comment]
The first option I would use some custom byte format thing (like tarkov doing) or just zlib compression
The second could be another WS but for all these unreliable packets.
